### ISSUE TITLE
New Version Won't Open: 1.3.3 (ppa:bit-team/stable, Debian unstable/bookworm, Ubuntu unstable/Lunar Lobster)

### ISSUE THREAD

**Issue by ryankingart** [bit-team/backintime#1401]
I upgraded to the latest version of Back in Time via the update manager, but when I try to open Back in Time, or Back in Time Root, it simply will not open. I tried re-installing it, and I also tried re-starting my computer, but that doesn't work either. The program will just not open.

My OS: Linux Mint 19.3 Cinnamon

Is there a possibility that the new version of back in time doesn't work on my OS? I know that my current Linux Mint version is old, and will be unsupported in april. So I plan on installing the latest version of Linux Mint soon. But I need to backup my files before I do that.

If anyone can help, I'd appreciate it.
Thanks!

**Comment by aryoda** [comment]
> I upgraded to the latest version of Back in Time via the update manager

Did you use [our PPA](https://github.com/bit-team/backintime#ubuntu-ppa) for that or another source?

Could you please add the output of

`backintime --debug`

and

`backintime --diagnostics`

here to make it possible for me to dig deeper into the problem?

It may just be a dependency issue (or even a too old python version)...

**Comment by ryankingart** [comment]
Hello. Thanks for the quick reply!

I installed the program with these commands:
`sudo add-apt-repository ppa:bit-team/stable`
`sudo apt-get update`
`sudo apt-get install backintime-qt`

**Here is the terminal output of: backintime --debug**

Traceback (most recent call last):
  File "/usr/share/backintime/common/backintime.py", line 28, in <module>
    import config
  File "/usr/share/backintime/common/config.py", line 45, in <module>
    import tools
  File "/usr/share/backintime/common/tools.py", line 37, in <module>
    from packaging.version import Version
ModuleNotFoundError: No module named 'packaging'

**Here is the terminal output of: backintime --diagnostics**

Traceback (most recent call last):
  File "/usr/share/backintime/common/backintime.py", line 28, in <module>
    import config
  File "/usr/share/backintime/common/config.py", line 45, in <module>
    import tools
  File "/usr/share/backintime/common/tools.py", line 37, in <module>
    from packaging.version import Version
ModuleNotFoundError: No module named 'packaging'

**Comment by aryoda** [comment]
This is a missing dependency. Please also install:

`sudo apt install python3-packaging`

and try it again (start GUI and in case of problems post the `backintime --debug` output here again.

**Comment by ryankingart** [comment]
That fixed it! The program opens now and seems to be working fine. Thank you so much for your help!

**Comment by aryoda** [comment]
Welcome! 

I think we have missing dependency and should fix this in our ppa (testing + stable)!

@Germar @buhtz I may be blind today: Where do we need to add the missing dependency so that our ppa is working correctly?

**Comment by aryoda** [comment]
@ryankingart I am re-openening the issue just for ourselves (= developers) to keep track of fixing the dependency declaration file(s)...

**Comment by ryankingart** [comment]
> @ryankingart I am re-openening the issue just for ourselves (= developers) to keep track of fixing the dependency declaration file(s)...

Sounds good. 👍‍

**Comment by Martii** [comment]
Same issue in:

``` console
$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 22.04.1 LTS
Release:        22.04
Codename:       jammy
```

... and ...

``` console
$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 20.04.5 LTS
Release:        20.04
Codename:       focal
```

> `sudo apt install python3-packaging`

That fixed it... my heart dropped when I saw this on a clients machine last night. Thanks for the work-around and keep up this awesome project. :smile: 

---

Sources btw:

<details>
  <summary>jammy</summary>
  <p>

``` console
$ apt-cache policy backintime-qt
backintime-qt:
  Installed: 1.3.3~jammy
  Candidate: 1.3.3~jammy
  Version table:
 *** 1.3.3~jammy 500
        500 https://ppa.launchpadcontent.net/bit-team/stable/ubuntu jammy/main amd64 Packages
        500 https://ppa.launchpadcontent.net/bit-team/stable/ubuntu jammy/main i386 Packages
        100 /var/lib/dpkg/status
     1.2.1-3ubuntu0.1 500
        500 http://us.archive.ubuntu.com/ubuntu jammy-updates/universe amd64 Packages
        500 http://us.archive.ubuntu.com/ubuntu jammy-updates/universe i386 Packages
     1.2.1-3 500
        500 http://us.archive.ubuntu.com/ubuntu jammy/universe amd64 Packages
        500 http://us.archive.ubuntu.com/ubuntu jammy/universe i386 Packages

$ apt-cache policy python3
python3:
  Installed: 3.10.6-1~22.04
  Candidate: 3.10.6-1~22.04
  Version table:
 *** 3.10.6-1~22.04 500
        500 http://us.archive.ubuntu.com/ubuntu jammy-updates/main amd64 Packages
        100 /var/lib/dpkg/status
     3.10.4-0ubuntu2 500
        500 http://us.archive.ubuntu.com/ubuntu jammy/main amd64 Packages

$ apt-cache policy python3-packaging
python3-packaging:
  Installed: 21.3-1
  Candidate: 21.3-1
  Version table:
 *** 21.3-1 500
        500 http://us.archive.ubuntu.com/ubuntu jammy/main amd64 Packages
        500 http://us.archive.ubuntu.com/ubuntu jammy/main i386 Packages
        100 /var/lib/dpkg/status
```

</p>
</details>

... and...

<details>
  <summary>focal</summary>
  <p>

``` console
$ apt-cache policy backintime-qt
backintime-qt:
  Installed: 1.3.3~focal
  Candidate: 1.3.3~focal
  Version table:
 *** 1.3.3~focal 500
        500 http://ppa.launchpad.net/bit-team/stable/ubuntu focal/main amd64 Packages
        500 http://ppa.launchpad.net/bit-team/stable/ubuntu focal/main i386 Packages
        100 /var/lib/dpkg/status
     1.2.1-2 500
        500 http://us.archive.ubuntu.com/ubuntu focal/universe amd64 Packages
        500 http://us.archive.ubuntu.com/ubuntu focal/universe i386 Packages

$ apt-cache policy python3
python3:
  Installed: 3.8.2-0ubuntu2
  Candidate: 3.8.2-0ubuntu2
  Version table:
 *** 3.8.2-0ubuntu2 500
        500 http://us.archive.ubuntu.com/ubuntu focal/main amd64 Packages
        100 /var/lib/dpkg/status

$ apt-cache policy python3-packaging
python3-packaging:
  Installed: 20.3-1
  Candidate: 20.3-1
  Version table:
 *** 20.3-1 500
        500 http://us.archive.ubuntu.com/ubuntu focal/main amd64 Packages
        500 http://us.archive.ubuntu.com/ubuntu focal/main i386 Packages
        100 /var/lib/dpkg/status
```

</p>
</details>

**Comment by buhtz** [comment]
> @Germar @buhtz I may be blind today: Where do we need to add the missing dependency so that our ppa is working correctly?

Information like this would be written into a `pyproject.toml` file these days when following the recommended python packaging standards. A file like this (or the infrastructure behind it) would then be used by each distribution and also from-upstream-installing-users. No need for any distro-specific files anymore in the repo.

But there is much more reorganizing work to do on file- and code-level before we can implement that.

**Comment by buhtz** [comment]
Just a sitenote because other users may report this problem in the next days.

At the time this Issue was reported the `python3-packaging` dependency was missing in Debian unstable (1.3.3-3) and because of that in Ubuntu unstable, too. But it was [reported](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1030051) at Debian and [fixed](https://salsa.debian.org/jmw/pkg-backintime/-/commit/a74e248394524e121bf1f2176cd4f2c9d74d591f) very quick before I even noticed this. :smile:

I assume it will take some days before this is uploaded to Debian/Ubuntu. It will appear there as `1.3.3-4`.

**Comment by aryoda** [comment]
Update: Fix is merged into `main` (PR #1402) and `dev` (PR #1403) now...

**Comment by tneo** [comment]
Installing the dependency python310-packaging on openSUSE resolves this issue. The fix will that apply to openSUSE as well?

**Comment by buhtz** [comment]
> The fix will that apply to openSUSE as well?

Do you mean if that will fix backintime in openSUSE, too? It seems to me that there is [no package in OpenSuse](https://software.opensuse.org/package/backintime).

**Comment by tneo** [comment]
Backintime is in the openSUSE-main repository. :)
So I'll assume the fix will make it as well.

**Comment by Martii** [comment]
Sources for...

<details>
  <summary>openSUSE Leap</summary>
  <p>

``` console
$ # Default updated openSUSE Leap (stable and equivalent to LTS), `lsb` installed, and `zypper update`

$ lsb_release -a
LSB Version:    n/a
Distributor ID: openSUSE
Description:    openSUSE Leap 15.4
Release:        15.4
Codename:       n/a

$ zypper search python3-packaging
Loading repository data...
Reading installed packages...

S | Name              | Summary                            | Type
--+-------------------+------------------------------------+--------
i | python3-packaging | Core utilities for Python packages | package

$ zypper info python3-packaging
Loading repository data...
Reading installed packages...


Information for package python3-packaging:
------------------------------------------
Repository     : Main Repository
Name           : python3-packaging
Version        : 20.3-1.9
Arch           : noarch
Vendor         : SUSE LLC <https://www.suse.com/>
Installed Size : 263.1 KiB
Installed      : Yes (automatically)
Status         : up-to-date
Source package : python-packaging-20.3-1.9.src
Upstream URL   : https://github.com/pypa/packaging
Summary        : Core utilities for Python packages
Description    :
    Core utilities for Python packages

$ zypper search python310-packaging
Loading repository data...
Reading installed packages...
No matching items found.

$ zypper info python310-packaging
Loading repository data...
Reading installed packages...


package 'python310-packaging' not found.

$ zypper search backintime
Loading repository data...
Reading installed packages...

S | Name            | Summary                                                 | Type
--+-----------------+---------------------------------------------------------+--------
  | backintime      | Backup tool for Linux inspired by the "flyback project" | package
  | backintime-lang | Translations for package backintime                     | package
  | backintime-qt   | Back In Time Qt5 GUI                                    | package

$ zypper info backintime
Loading repository data...
Reading installed packages...


Information for package backintime:
-----------------------------------
Repository     : Main Repository
Name           : backintime
Version        : 1.3.1-bp154.1.27
Arch           : noarch
Vendor         : openSUSE
Installed Size : 642.0 KiB
Installed      : No
Status         : not installed
Source package : backintime-1.3.1-bp154.1.27.src
Upstream URL   : https://github.com/bit-team/backintime
Summary        : Backup tool for Linux inspired by the "flyback project"
Description    :
    Back In Time is a backup tool for Linux inspired by the "flyback project".

    It provides a command line client 'backintime' and a Qt5 GUI 'backintime-qt'
    both written in Python3.

    You only need to specify 3 things:
        * where to save snapshots;
        * what folders to backup; and
        * backup frequency (manual, every hour, every day, every month).
```

</p>
</details>

... and ...

<details>
  <summary>openSUSE Tumbleweed</summary>
  <p>

``` console
$ # Default openSUSE Tumbleweed (Possibly unstable and equivalent to a rolling distro), `lsb` installed, and `zypper update`

$ lsb_release -a
LSB Version:    core-2.0-noarch:core-3.2-noarch:core-4.0-noarch:core-2.0-x86_64:core-3.2-x86_64:core-4.0-x86_64:desktop-4.0.fake-amd64:desktop-4.0.fake-noarch:graphics-2.0-amd64:graphics-2.0-noarch:graphics-3.2-amd64:graphics-3.2-noarch:graphics-4.0.fake-amd64:graphics-4.0.fake-noarch
Distributor ID: openSUSE
Description:    openSUSE Tumbleweed
Release:        20230131
Codename:       n/a

$ zypper search python3-packaging
Loading repository data...
Reading installed packages...
No matching items found.

$ zypper search python310-packaging
Loading repository data...
Reading installed packages...

S | Name                | Summary                            | Type
--+---------------------+------------------------------------+--------
  | python310-packaging | Core utilities for Python packages | package

$ zypper info python310-packaging
Loading repository data...
Reading installed packages...


Information for package python310-packaging:
--------------------------------------------
Repository     : Main Repository (OSS)
Name           : python310-packaging
Version        : 23.0-1.1
Arch           : noarch
Vendor         : openSUSE
Installed Size : 147.1 KiB
Installed      : No
Status         : not installed
Source package : python310-packaging-23.0-1.1.src
Upstream URL   : https://packaging.pypa.io/
Summary        : Core utilities for Python packages
Description    :
    Reusable core utilities for various Python Packaging interoperability specifications.

    This library provides utilities that implement the interoperability specifications
    which have clearly one correct behaviour (eg: PEP 440) or benefit greatly from having
    a single shared implementation (eg: PEP 425).

$ zypper search backintime
Loading repository data...
Reading installed packages...

S | Name            | Summary                                                 | Type
--+-----------------+---------------------------------------------------------+--------
  | backintime      | Backup tool for Linux inspired by the "flyback project" | package
  | backintime-lang | Translations for package backintime                     | package
  | backintime-qt   | Back In Time Qt5 GUI                                    | package

$ zypper info backintime
Loading repository data...
Reading installed packages...


Information for package backintime:
-----------------------------------
Repository     : Main Repository (OSS)
Name           : backintime
Version        : 1.3.3-1.1
Arch           : noarch
Vendor         : openSUSE
Installed Size : 695.4 KiB
Installed      : No
Status         : not installed
Source package : backintime-1.3.3-1.1.src
Upstream URL   : https://github.com/bit-team/backintime
Summary        : Backup tool for Linux inspired by the "flyback project"
Description    :
    Back In Time is a backup tool for Linux inspired by the "flyback project".

    It provides a command line client 'backintime' and a Qt5 GUI 'backintime-qt'
    both written in Python3.

    You only need to specify 3 things:
        * where to save snapshots;
        * what folders to backup; and
        * backup frequency (manual, every hour, every day, every month).
```

</p>
</details>

**Comment by aryoda** [comment]
> Backintime is in the openSUSE-main repository. :)
> So I'll assume the fix will make it as well.

**TLDR:** Please open an downstream issue (= at the web site of your distro's package maintainer), we cannot fix this.

**Long version:**

My "fix" here does only add the missing dependency to our Ubuntu PPA (personal package archive) which BTW is **no** official Ubuntu repository.

The "fix" **does not** change any source code of BiT.

To add the missing dependency to each installation package of each distro (incl. OpenSuse) the responsible package maintainer must be informed (e.g. by opening an issue in their distro packaging web site) as only the package maintainer can do this.

In the future I think we need

1. a `changelog` file for package maintainers to clearly indicate dependency (and other) changes in new BiT releases
2. and could maintain a list of all known package maintainers (which is hard to track for us) to direct users with packaging-related issues to their corresponding package maintainer

**Any help to compile and maintain such a package maintainer list is highly appreciated (please send a PR)!**

I propose to close this issue since we have fixed the missing dependency in the only repo where we are the package maintainers ourself ([Ubuntu PPA](https://launchpad.net/~bit-team))...

**Comment by tneo** [comment]
I have sent the package maintainer a message to make him aware of this issue.

**Comment by buhtz** [comment]
> Backintime is in the openSUSE-main repository. :) So I'll assume the fix will make it as well.

Just for my learning. Maybe I misunterprete the link. Can you link to the opensuse-packag-website where I can see informations like this. The link I used just showed my "Versuchspackete" (testing?) and "Community-Pakete" (community).

> I have sent the package maintainer a message to make him aware of this issue.

If this is a public available ticket please link it here.

**Comment by buhtz** [comment]
> In the future I think we need
> 
>     1. a `changelog` file for package maintainers to clearly indicate dependency (and other) changes in new BiT releases

We do have a changelog. IMHO no need for redundancy. A good/motivated distro maintainer take care of such a file.

And especially dependency are nothing that need to be written down for humans. This is something for the machines only. I mean if the repo is setup correctly fitting to the pythong packaging standards (e.g. using a `pyproject.toml`) all dependencies are "coded" in there. For example the Debian packaging tools do use that file when creating a deb file.

It is on my todo list and I will show you kind of a demo project imitating the BIT repo and all its needs.

>     2. and could maintain a list of all known package maintainers (which is hard to track for us) to direct users with packaging-related issues to their corresponding package maintainer

We have `bit-dev` they can subsribe to if they are interested.
We could create `bit-announce` but IMHO there is no real need for this.
Most distros do monitor the upstream repos in a automatic way. I know the debian maintainers will get informed about each new release everytime we set a `v*` tag in the `main` branch.

> I propose to close this issue

Agree.

**Comment by Martii** [comment]
@buhtz 

> Can you link to the opensuse-packag-website where I can see informations like this.

It's not very robust on the search but found https://download.opensuse.org/tumbleweed/repo/oss/noarch/ and https://download.opensuse.org/distribution/leap/15.4/repo/oss/noarch/ as an alternative. Just type in the search. The only thing you need to know is what "category" *(OSS, Non-OSS, noarch, etc.)* it is... otherwise the search, here at least, doesn't work at the top levels mentioned at https://en.opensuse.org/Package_repositories e.g. https://download.opensuse.org/tumbleweed/repo/ 

I just installed a couple of VM's to get the exact information [above](https://github.com/bit-team/backintime/issues/1401#issuecomment-1412691207) for reference.

P.S. Apologies for the edits but something is screwy with klipper atm or the mouse... grr. :smile_cat:

**Comment by buhtz** [comment]
FYI

Debian fixed it as a [patch](https://salsa.debian.org/jmw/pkg-backintime/-/commit/c12e197ced44392e6636259012d41d3e285ad5ce#note_379754) on their side after I opened a [ticket](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1030506).

**Comment by buhtz** [comment]
Related to PR #1410 . When merged this Issue should be fixed.
